Simplenote/src/main/java/com/automattic/simplenote/authentication/SignupFragment.java

```diff
@@ -0,0 +1,264 @@
+    }
+
+    private void listenToEmailChanges(EditText emailEditText, final Button signupButton) {
+        emailEditText.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void afterTextChanged(Editable s) {
+                setButtonState(signupButton, s);
+            }
+
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+            }
+        });
+    }
+
+    private void listenToSignupClick(Button signupButton, final EditText emailEditText) {
+        signupButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (NetworkUtils.isNetworkAvailable(requireContext())) {
+                    showProgressDialog();
+                    signupUser(emailEditText.getText().toString());
+                } else {
+                    showDialogError(getString(R.string.simperium_dialog_message_network));
+                }
+            }
+        });
+    }
+
+    private void showProgressDialog() {
+        progressDialogFragment =
+            ProgressDialogFragment.newInstance(getString(R.string.simperium_dialog_progress_signing_up));
+        progressDialogFragment.setStyle(DialogFragment.STYLE_NO_TITLE, R.style.Simperium);
+        progressDialogFragment.show(requireFragmentManager(), ProgressDialogFragment.TAG);
+    }
+
+    private void hideDialogProgress() {
+        if (progressDialogFragment != null && !progressDialogFragment.isHidden()) {
+            progressDialogFragment.dismiss();
+            progressDialogFragment = null;
+        }
+    }
+
+    private void signupUser(String email) {
+        new OkHttpClient()
+            .newBuilder()
+            .readTimeout(TIMEOUT_SECS, TimeUnit.SECONDS)
+            .build()
+            .newCall(buildCall(email))
+            .enqueue(buildCallback(email));
+    }
+
+    private Request buildCall(String email) {
+        return new Request.Builder()
+            .url(buildUrl())
+            .post(buildJsonBody(email))
+            .header(ACCEPT_LANGUAGE, getLanguage())
+            .build();
+    }
+
+    private RequestBody buildJsonBody(String email) {
+        JSONObject json = new JSONObject();
+        try {
+            json.put(JSON_USERNAME, email);
+        } catch (JSONException e) {
+            throw new IllegalArgumentException("Cannot construct json with supplied email: " + email);
+        }
+        return RequestBody.create(JSON_MEDIA_TYPE, json.toString());
+    }
+
+    private HttpUrl buildUrl() {
+        return new HttpUrl.Builder()
+            .scheme(HTTP_SCHEME)
+            .host(HTTP_HOST)
+            .addPathSegment(SIMPLENOTE_ACCOUNT_PATH)
+            .addPathSegment(SIMPLENOTE_REQUEST_SIGNUP_PATH)
+            .build();
+    }
```

Comment:
this looks like it would be hard to search for in the source code (since the URL we want doesn't exist anywhere as a whole piece) and it's also confusing what it's really hitting, having to jump around in the file to understand. what are your thoughts on making this less abstract and using `HttpUrl.parse()` since it's all static anyway?

```java
HttpUrl.parse("https://app.simplenote.com/account/request-signup");
```

Reply:
I agree it confusing a bit, especially with multiple path segments, what I would do is passing the path as a single call using `addPathSegments`, but keep the host separated, and this to make it easier using different hosts, if we want to hit a different server for the development builds, we can then have the host in the `gradle.properties` like the simperium app id.
We were actually planning to discuss this point with you, because right now, the external contributors have no way to test the signup process, because we only offer the simperium server `history-analyst-dad` for debugging, but as it's not connected to `app.simplenote.com`, the signup won't work, I think it would be awesome if we can have a different development server connected to this simperium entity.

Reply:
we probably won't have a signup server available for community contributors, unfortunately. if someone wants to fake the response they would be best to use any available network proxy/faker tool

Reply:
👋 @dmsnell and @hichamboushaba !

I agree with both of you and I was thinking the same on the multiple `addPathSegment(...)` build calls. So, I think it is better if we group those and leave the other parts as is. I also agree with @hichamboushaba on his points. This kind of API call on Android, using `OkHttp` directly are dated, libraries like [Retrofit](https://square.github.io/retrofit/), which are the go to libraries for that matter, for many years now, are doing the same, but by using annotations instead to expose the segment path, the rest is abstracted elsewhere. Thus, I choose the same path in order to align with that.

PS: If you really feel strong about that we can definitely change that to a full path: `https://app.simplenote.com/account/request-signup`

Reply:
This is now done here: 96eeb89cddb0af23a644d80ec83f6b2855f4dda2